gitops/predicates_test.go

```diff
@@ -68,14 +68,14 @@ var _ = Describe("Predicates", Ordered, func() {
 		}
 		ctx := context.Background()
 
-		Expect(k8sClient.Create(ctx, bindingUnknownStatus)).Should(Succeed())
+		Expect(k8sClient.Create(ctx, bindingFalseStatus)).Should(Succeed())
 		Expect(k8sClient.Create(ctx, bindingTrueStatus)).Should(Succeed())
 
 		// Set the binding statuses after they are created
-		bindingUnknownStatus.Status.ComponentDeploymentConditions = []metav1.Condition{
+		bindingFalseStatus.Status.ComponentDeploymentConditions = []metav1.Condition{
 			{
```

Comment:
I think the condition can be empty after it's created. Then the BindingDeploymentStatusConditionType condition is set after the snapshot is deployed.

Reply:
I agree with you. But if we keep it empty, we won't be able to test the case when a SEB's BindingDeploymentStatusConditionType condition changes from False to True.

Reply:
From Kruno's comment, if I understand correctly, the real situation should be from an empty status to  AllComponentsDeployed status condition being set.

Reply:
I think it would be good to test both cases - when the `BindingDeploymentStatusConditionType` changes from False to True as well as when the condition doesn't exist and changes to `True`.

Reply:
Thanks all! I've updated the code to allow both scenarios when the status changes from "False to True" and from "Nothing to True".

Reply:
Resolving this discussion to unblock the PR as the points have been addressed. We can follow up on slack if anything needs to be cleared up.